### PR TITLE
[UnitTests] Added cuDNN to default test targets

### PR DESCRIPTION
**Pull request by Lunderberg**
Some unit tests explicitly test cudnn in addition to tvm.testing.enabled_targets().  This moved the cudnn checks into the same framework as all other targets, and adds it to the default list of targets to be run.  Also, added `@tvm.testing.requires_cudnn` for tests specific to cudnn.

In principle, this should pass CI, since any non-cudnn tests can fall back to the cuda implementation, but "should" is a bit of a magic word.

### CODE REVIEW

**Comment by Lunderberg** [comment]
Updated to fix pylint error.

**Comment by Lunderberg** [comment]
Current failures.

* `test_conv2d_nchw` doesn't support asymmetric padding on cudnn.  Can add an explicit `pytest.xfail` on that test case once #8433 is merged.
* `test_batch_matmul` errors out for dynamic batch matmul not being supported on cuda.  Changed check from direct comparison with the string `"cuda"` to a check on the `target.kind.name`.

File: python/tvm/testing.py
Comment (comaniac) [inline python/tvm/testing.py:384]: nit
```suggestion
    if not target_names:
        target_names = DEFAULT_TEST_TARGETS
```

**Review by comaniac — APPROVED** [review]
It would be better to also let a Vulkan folk review the changes in `target_kind.cc`.

File: python/tvm/testing.py
Comment (Lunderberg) [reply python/tvm/testing.py]: Good call, and updated.

**Comment by Lunderberg** [comment]
Will need rebasing on main to resolve conflicts with #8602 (merged) and #8542 (waiting on CI).  #8542 is the more important one, so this PR can wait until it is done.

**Comment by Lunderberg** [comment]
Hopefully the last update needed here, rebased to main and fixed an issue with the conv2d_cudnn handling of dilation.

**Pull request merged**
